src/daemon/rpmostreed-os.c

```diff
@@ -77,6 +81,101 @@ sysroot_changed (RpmostreedSysroot *sysroot,
     g_warning ("%s", local_error->message);
 }
 
+static gboolean
+os_authorize_method (GDBusInterfaceSkeleton *interface,
+                     GDBusMethodInvocation  *invocation)
+{
+  RpmostreedOS *self = RPMOSTREED_OS (interface);
+  const gchar *method_name = g_dbus_method_invocation_get_method_name (invocation);
+  const gchar *sender = g_dbus_method_invocation_get_sender (invocation);
+  GVariant *parameters = g_dbus_method_invocation_get_parameters (invocation);
+  const gchar *action = NULL;
+  gboolean authorized = FALSE;
+
+  if (g_strcmp0 (method_name, "GetDeploymentsRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "GetCachedDeployRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "DownloadDeployRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "GetCachedUpdateRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "DownloadUpdateRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "GetCachedRebaseRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "DownloadRebaseRpmDiff") == 0)
```

Comment:
This looks OK, but I wonder if there's a best practice for embedding the "action binding" with the method itself?  Maybe something like including it in an annotation in the XML and having some way to get at that?  I think I saw that systemd defines structs for its methods, but I'm not sure how easy that is to do with GDBus.